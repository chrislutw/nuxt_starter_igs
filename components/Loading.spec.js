import { shallowMount } from '@vue/test-utils'
import Loading from '~/components/Loading'
import NuxtLink from '~/.nuxt/components/nuxt-link'

describe('index', () => {
  beforeAll(() => {
    // index.components.NuxtLink = NuxtLink
  })

  test('mounts properly', () => {
    const wrapper = shallowMount(Loading)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(Loading)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('calls handleClick on click', () => {
    const wrapper = shallowMount(Loading)
    const handleClickMock = jest.fn()
    wrapper.vm.loading = true
    wrapper.setMethods({
      start: handleClickMock
    })
    wrapper.vm.start()
    expect(handleClickMock).toHaveBeenCalled()
  })
})
