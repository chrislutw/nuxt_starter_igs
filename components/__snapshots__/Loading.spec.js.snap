// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `undefined`;
